File: experiments/020-PythonVsJava/cells/python-pytest_1-Empty_free/2019-2c-parcial2-masked/20260921-204513-77784/exercise/java/src/main/java/surveycrawler/Point.java
package surveycrawler;

public class Point {

    private final int x;
    private final int y;

    public Point(int x, int y) {
        this.x = x;
        this.y = y;
    }

    public int x() {
        return x;
    }

    public int y() {
        return y;
    }

    public Point plus(Point aPoint) {
        return new Point(x + aPoint.x, y + aPoint.y);
    }

    @Override
    public boolean equals(Object anObject) {
        if (!(anObject instanceof Point)) {
            return false;
        }
        Point aPoint = (Point) anObject;
        return x == aPoint.x && y == aPoint.y;
    }

    @Override
    public int hashCode() {
        return 31 * x + y;
    }

    @Override
    public String toString() {
        return x + "@" + y;
    }
}

File: experiments/020-PythonVsJava/cells/python-pytest_1-Empty_free/2019-2c-parcial2-masked/20260921-202806-59410/output/project/src/surveycrawler/point.py
class Point:

    def __init__(self, x, y):
        self._x = x
        self._y = y

    def x(self):
        return self._x

    def y(self):
        return self._y

    def plus(self, a_point):
        return Point(self._x + a_point._x, self._y + a_point._y)

    def times(self, a_number):
        return Point(self._x * a_number, self._y * a_number)

    def inverted(self):
        return self.times(-1)

    def __eq__(self, an_object):
        if not isinstance(an_object, Point):
            return False
        a_point = an_object
        return self._x == a_point._x and self._y == a_point._y

    def __hash__(self):
        return 31 * self._x + self._y

    def __repr__(self):
        return str(self._x) + "@" + str(self._y)
